Source/DafnyLanguageServer/Language/BoogieExtensions.cs

```diff
@@ -49,5 +52,27 @@ public static class BoogieExtensions {
     public static Position ToLspPosition(int boogieLine, int boogieColumn) {
       return new Position(boogieLine + LineOffset, boogieColumn + ColumnOffset);
     }
+
+    public static IToken ToToken(this Position position, string document) {
+      try {
+        return new Token() {
+          line = position.Line - LineOffset,
+          col = position.Character - ColumnOffset,
+          val = "",
+          pos = position.ToAbsolutePosition(document)
```

Comment:
Calling `ToAbsolutePosition` seems like a performance problem. Each time the user changes their cursor to a new line, `textDocument/codeAction` is called and this triggers an iteration over the entire document on the server. I think we should avoid doing any document sized operations. Wouldn't it be better not to have `PositionExtensions.ToAbsolutePosition` at all and force clients that want to access text based on ranges to work through `DocumentTextBuffer` ?

Reply:
This call will only happen when the user places their caret on a line with a verification diagnostic.
However, I refactored the code now so that it will call `ToAbsolutePösition` only when accepting a Quick fix